test/services/creates_user_discussion_post_test.rb

```diff
@@ -47,6 +49,19 @@ class CreatesUserDiscussionPostTest < ActiveSupport::TestCase
     CreatesUserDiscussionPost.create!(iteration, user, "foooebar")
   end
 
+  test "does not notify non-current mentors" do
+    iteration = create :iteration
+    solution = iteration.solution
+    user = solution.user
+
+    non_current_mentor = create :user
```

Comment:
Same, I'd go with `legacy_commenter` to distinguish between those who actually signed up to be mentors, and those who have solution mentorships because they commented in v1.